pandas/_libs/tslibs/src/datetime/np_datetime.c

```diff
@@ -331,6 +331,34 @@ int cmp_npy_datetimestruct(const npy_datetimestruct *a,
 
     return 0;
 }
+/*
+* Returns the offset from utc of the timezone.
+* If the passed object is timezone naive, or if extraction
```

Comment:
I would advise against having NULL mean potentially two different things. I think it would be cleanest if NULL explicitly signaled an error, so that handlers know to either pass that back up the stack or check the error message and handle accordingly. We can just add checks for None in callers of this right?

https://docs.python.org/3/c-api/exceptions.html#exception-handling

Reply:
Alternately you can move the hasattr check and do that before calling the function. That is likely the most explicit way of handling